app/uk/gov/hmrc/gform/sharedmodel/graph/DependencyGraph.scala

```diff
@@ -147,28 +144,32 @@ object DependencyGraph {
       formModel.allFormComponents.toSet
         .flatMap(edges) ++ includeIfs ++ componentIncludeIfs ++ validIfs ++ sections
 
-    val allFcIds: Set[ModelComponentId] = allEdges.flatMap { diEdge =>
-      diEdge.collectFirst {
-        case GraphNode.Expr(FormCtx(fcId)) => fcId.modelComponentId
-        case GraphNode.Simple(fcId)        => fcId.modelComponentId
-      }
-    }
+    val atlFieldsBaseIds: Set[BaseComponentId] =
+      formModel.brackets.addToListBrackets.flatMap(_.source.pages.toList.flatMap(_.fields.map(_.baseComponentId))).toSet
 
     // This represents references to atl fields made outside of atl.
     val addToListEdges: Iterable[DiEdge[GraphNode]] =
-      allFcIds
+      formModel.brackets.addToListBrackets
+        .flatMap(_.iterations.toList.flatMap(_.singletons.toList.map(x => x.singleton.page)))
+        .flatMap(_.fields)
         .groupBy(_.baseComponentId)
-        .collect {
-          case (k, v) if atlFieldsBaseIds(k) =>
-            v.map(_.indexedComponentId)
-              .collect { case x: IndexedComponentId.Indexed =>
-                x
-              }
-              .map { v =>
-                GraphNode.Simple(ModelComponentId.pure(IndexedComponentId.pure(k)).toFormComponentId) ~> GraphNode.Expr(
-                  FormCtx(ModelComponentId.pure(v).toFormComponentId)
-                )
-              }
+        .collect { case (k, fcs) =>
+          fcs.flatMap { fc =>
+            (fc match {
+              case f @ IsRevealingChoice(revealingChoice) if atlFieldsBaseIds(k) =>
```

Comment:
This special `IsRevealingChoice` case is awkward.
I have provided slightly different implementation [here](https://github.com/hmrc/gform-frontend/commit/ce528a587faf0abfca6d56849162ae9266b1f448)

Reply:
I cherry-picked your PR. :+1: